File: src/main/java/casa_de_cha/model/Lote.java
package casa_de_cha.model;

import lombok.AllArgsConstructor;
import lombok.NoArgsConstructor;

import javax.persistence.*;
import java.io.Serial;
import java.io.Serializable;
import java.util.Date;

@Entity
@NoArgsConstructor
@AllArgsConstructor
@SequenceGenerator(name = "seq_lote", sequenceName = "seq_lote", allocationSize = 1)
public class Lote implements Serializable {

    @Serial
    private static final long serialVersionUID = 1L;
    @Id
    @GeneratedValue(strategy = GenerationType.SEQUENCE, generator = "seq_lote")
    private int id_Lote;
    @Column
    private Date dt_validade;
    @Column
    private Date dt_registro;
    @Column
    private int qtd_lote;

    private int id_produto;

    public int getId_Lote() {
        return id_Lote;
    }

    public void setId_Lote(int id_Lote) {
        this.id_Lote = id_Lote;
    }

    public Date getDt_validade() {
        return dt_validade;
    }

    public void setDt_validade(Date dt_validade) {
        this.dt_validade = dt_validade;
    }

    public Date getDt_registro() {
        return dt_registro;
    }

    public void setDt_registro(Date dt_registro) {
        this.dt_registro = dt_registro;
    }

    public int getQtd_lote() {
        return qtd_lote;
    }

    public void setQtd_lote(int qtd_lote) {
        this.qtd_lote = qtd_lote;
    }

    public int getProduto() {
        return id_produto;
    }

    public void setProduto(int id_produto) {
        this.id_produto = id_produto;
    }

}

File: front-end/casa_de_cha/src/core/model/Lote.ts
import { format } from 'date-fns';

export default class Lote {
    private _id: string
    private _qtd_lote: number
    private _dt_validade: string
    private _dt_registro: string
    private _id_produto: string

    constructor(id: string, qtd_lote: number, dt_validade: string, dt_registro: string, id_produto: string){
        this._qtd_lote = qtd_lote
        this._dt_registro = dt_registro
        this._dt_validade = dt_validade
        this._id = id
        this._id_produto = id_produto
    }

    static vazio() {
        const currentDate = new Date();

        return new Lote('', 0, '', format(currentDate, 'dd/MM/yyyy'), '');
    }

    get id() {
        return this._id
    }

    get qtd_lote() {
        return this._qtd_lote
    }

    get dt_validade() {
        return this._dt_validade
    }

    get dt_registro() {
        return this._dt_registro
    }

    get id_produto() {
        return this._id_produto
    }
}
